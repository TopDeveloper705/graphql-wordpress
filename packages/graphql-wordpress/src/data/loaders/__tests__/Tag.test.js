import getTagLoaders from 'data/loaders/Tag';

const Tag = getTagLoaders();

describe('Test Tag loader', () => {
  test('Load a tag', async () => {
    const tag = await Tag.load(13);
    expect(tag.getID()).toMatchSnapshot();
    expect(tag).toMatchSnapshot();
  });

  test('Load a tag by slug', async () => {
    const tag = await Tag.loadBySlug('dirty-projectors');
    expect(tag).toMatchSnapshot();
  });

  test('Load tags', async () => {
    const tag = await Tag.loadMany([13, 17]);
    expect(tag).toMatchSnapshot();
  });
});
