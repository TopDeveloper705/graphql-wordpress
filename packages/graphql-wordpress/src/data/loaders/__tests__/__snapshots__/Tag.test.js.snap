// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Tag loader Load a tag 1`] = `"VGFnOjEz"`;

exports[`Test Tag loader Load a tag 2`] = `
Tag {
  "id": 13,
}
`;

exports[`Test Tag loader Load a tag by slug 1`] = `
Tag {
  "slug": "dirty-projectors",
}
`;

exports[`Test Tag loader Load tags 1`] = `
Array [
  Tag {
    "id": 13,
  },
  Tag {
    "id": 17,
  },
]
`;
